sql/timeseries/a11yScores.sql

```diff
@@ -1,32 +1,37 @@
 #standardSQL
-CREATE TEMPORARY FUNCTION getA11yScore(report STRING)
-RETURNS FLOAT64
+# Lighthouse changed format of scores in v3.0.0 released in July 2018 so handle old with a UDF
+CREATE TEMPORARY FUNCTION getA11yScore(reportCategories STRING)
+RETURNS FLOAT64 DETERMINISTIC
 LANGUAGE js AS """
-  $=JSON.parse(report);
-  if ($.reportCategories) {
-    return $.reportCategories.find(i => i.name === 'Accessibility').score;
-  } else {
-    return $.categories.accessibility.score * 100;
+  $=JSON.parse(reportCategories);
+  if($) {
+    return $.find(i => i.name === 'Accessibility').score;
   }
 """;
 
 SELECT
-  SUBSTR(_TABLE_SUFFIX, 0, 10) AS date,
-  UNIX_DATE(CAST(REPLACE(SUBSTR(_TABLE_SUFFIX, 0, 10), '_', '-') AS DATE)) * 1000 * 60 * 60 * 24 AS timestamp,
-  IF(ENDS_WITH(_TABLE_SUFFIX, 'desktop'), 'desktop', 'mobile') AS client,
-  ROUND(APPROX_QUANTILES(getA11yScore(report), 101)[OFFSET(11)], 2) AS p10,
-  ROUND(APPROX_QUANTILES(getA11yScore(report), 101)[OFFSET(26)], 2) AS p25,
-  ROUND(APPROX_QUANTILES(getA11yScore(report), 101)[OFFSET(51)], 2) AS p50,
-  ROUND(APPROX_QUANTILES(getA11yScore(report), 101)[OFFSET(76)], 2) AS p75,
-  ROUND(APPROX_QUANTILES(getA11yScore(report), 101)[OFFSET(91)], 2) AS p90
-FROM
-  `httparchive.lighthouse.*`
-WHERE
-  report IS NOT NULL
+  date,
+  UNIX_DATE(CAST(REPLACE(date, '_', '-') AS DATE)) * 1000 * 60 * 60 * 24 AS timestamp,
+  client,
+  ROUND(APPROX_QUANTILES(score, 1000)[OFFSET(100)], 2) AS p10,
+  ROUND(APPROX_QUANTILES(score, 1000)[OFFSET(250)], 2) AS p25,
+  ROUND(APPROX_QUANTILES(score, 1000)[OFFSET(500)], 2) AS p50,
+  ROUND(APPROX_QUANTILES(score, 1000)[OFFSET(750)], 2) AS p75,
+  ROUND(APPROX_QUANTILES(score, 1000)[OFFSET(900)], 2) AS p90
+FROM (
+  SELECT
+    SUBSTR(_TABLE_SUFFIX, 0, 10) AS date,
+    IF(ENDS_WITH(_TABLE_SUFFIX, 'desktop'), 'desktop', 'mobile') AS client,
+    IFNULL(CAST(JSON_EXTRACT(report, '$.categories.accessibility.score') AS FLOAT64) * 100,getA11yScore(JSON_EXTRACT(report, '$.reportCategories'))) AS score
```

Comment:
Should the second value also be *100?
```suggestion
    IFNULL(CAST(JSON_EXTRACT(report, '$.categories.accessibility.score') AS FLOAT64) * 100, getA11yScore(JSON_EXTRACT(report, '$.reportCategories')) * 100) AS score
```

Reply:
No it shouldn't, as per the old UDF:

https://github.com/HTTPArchive/bigquery/blob/e316b8f40bcea6fe47b4b214604a415a3d334d1e/sql/timeseries/a11yScores.sql#L6-L10